src/components/Settings/Settings.tsx

```diff
@@ -26,72 +25,67 @@ function SettingsCard() {
     dispatch(updateSettings(updates))
   }
 
-  function toggleExpand() {
-    setExpanded(!expanded)
+  function handleToggleButtonClick() {
+    setIsToggleButtonExpanded(!isToggleButtonExpanded)
   }
 
+  const renderBaseOptions = baseOptions.map(({ label, placeholder, value }) => (
+    <FormItem
+      key={value as string}
+      label={label}
+      placeholder={placeholder}
+      value={
+        (value in updates ? updates[value] : settings[value]) as SettingName
+      }
+      onChange={(e) => changeSetting(value, e.target.value)}
+    />
+  ))
+
+  const renderAdvancedOptions = advancedOptions.map(
+    ({ label, placeholder, value }) => (
+      <FormItem
+        key={value}
+        label={label}
+        placeholder={placeholder}
+        value={
+          (value in updates ? updates[value] : settings[value]) as SettingName
+        }
+        onChange={(e) => changeSetting(value as SettingName, e.target.value)}
+      />
+    )
+  )
+
   return (
-    <div className="mb-4">
-      <div className="border-bottom">
-        <h6 className="m-0">Connection Preferences</h6>
-      </div>
-      <div>
-        <div>
-          <div>
-            {baseOptions.map(({ label, placeholder, value }) => (
-              <FormItem
-                key={value as string}
-                label={label}
-                placeholder={placeholder}
-                value={
-                  (value in updates
-                    ? updates[value]
-                    : settings[value]) as SettingName
-                }
-                onChange={(e) => changeSetting(value, e.target.value)}
-              />
-            ))}
-          </div>
-          <div>
-            <div>
-              <strong
-                aria-controls="collapsed-form"
-                aria-expanded={expanded}
-                onClick={toggleExpand}
-                className="text-primary d-inline-block mb-3 cursor-pointer"
-              >
-                Advanced{" "}
-                <i className="material-icons">
-                  {expanded ? "arrow_drop_up" : "arrow_drop_down"}
-                </i>
+    <div className=" bg-white  rounded-md">
+      <h6 className="p-4 border-b text-gray-600">Connection Preferences</h6>
+      <div className="p-4">
+        <div className="grid grid-cols-2 gap-4 mb-4">{renderBaseOptions}</div>
+        <div className="flex mb-4 justify-between items-center">
+          <ToggleButton
+            isExpanded={isToggleButtonExpanded}
+            onToggleButtonClick={handleToggleButtonClick}
+          />
+          <button
```

Comment:
The button component inside common folder is not used since its prop variant="primary" is not exactly rendering the primary color of the app. Hence used button element with tailwind classes. In future we can change to common button component when the variant="primary" is working as expected.